notes/2020-10-02.md

```diff
@@ -30,52 +30,58 @@ ca_dds_df
 ```{code-cell} ipython3
 print(ca_dds_df)
 ```
-
+we can do df.index to look at what the index is in our dataframe
 ```{code-cell} ipython3
 ca_dds_df.index
 ```
-
+columns are tuple values. We can index to the first column as below. As we can see by running this code, this column has three names initially. 
 ```{code-cell} ipython3
 ca_dds_df.columns[0]
 ```
-
+By the following code we can assign the first column as the index of original dataframe. If we assign "False" to the "inplace", it will make the changes in a new dataframe.
 ```{code-cell} ipython3
 ca_dds_df.set_index(ca_dds_df.columns[0],inplace=True)
 ```
-
+The updated dataframe looks like below. We still have the name of index inappropriate and aslo there are empty rows. 
 ```{code-cell} ipython3
 ca_dds_df
 ```
-
+we can drop rows with "all" or "any",  "NaN" values. We would choose "all", since if we choose "any", we will loose many rows here and we do not want it.
 ```{code-cell} ipython3
 ca_dds_df.dropna(how='all',inplace=True)
 ```
-
+We can view the index by the following code. Remember that this is not a regular column anymore since we assigned it to be our dataframe index.
 ```{code-cell} ipython3
 ca_dds_df.index
 ```
-
+Now lets rename the index to a more meaningful name as below.
 ```{code-cell} ipython3
 ca_dds_df.index.rename('Age Cohort',inplace=True)
 ```
+But we still have multi-row headers which should get corrected. We also want to have columns for "Age", "Race", "Gender", "Mean" and "Count".
 
+
+
+Let's flip the dataframe and unstack the index. For this purpose, we should unstack it in two levels of "Race" and "Gender".  After unstacking we want to flip it again and finally reset the index.
 ```{code-cell} ipython3
 ca_dds_df.T.unstack(level= [0,1]).T.reset_index()
 ```
```

Comment:
For annotation, this can be re-split into multiple cells. Also add a bit more description of what we saw and look for at each step
````suggestion
Let's flip the dataframe
```{code-cell} ipython3
ca_dds_df.T
```
  unstack the index. For this purpose, we should unstack it in two levels of "Race" and "Gender".  
```{code-cell} ipython3
ca_dds_df.T.unstack(level= [0,1])
```
After unstacking we want to flip it again and finally reset the index.
```{code-cell} ipython3
ca_dds_df.T.unstack(level= [0,1]).T.reset_index()
```
````